src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -303,6 +303,20 @@ public function getRemovedIds(): array
         return [];
     }
 
+    /**
+     * Gets parameters that existed at compile time.
+     */
+    public function getRemovedParameters(): array
+    {
+        return array_fill_keys(
```

Comment:
I'd do it differently: this method should return `[]`, and ContainerBuilder should return `$this->removedParameters`, exactly as we do for removed ids. The convention itself (the leading dot) doesn't need to be known by ContainerBuilder.